src/system_identity.rs

```diff
@@ -0,0 +1,319 @@
+pub mod client_input;
+mod environment;
+mod generator;
+mod iam_client;
+mod output_platform;
+
+#[derive(Debug, Clone, Default)]
+pub struct SystemIdentity {
+    pub name: String,
+    pub client_id: String,
+    pub pub_key: Vec<u8>,
```

Comment:
would make sense to make this private. IIRC a SI might not have a keypair  but a client secret, and we might want to model that WDYT

Reply:
I thought that the SI generation will always produce a key pair? Not sure how this could be private if all the fields would be needed by some external system for e.g. persisting into the `OutputPlatform` (filesystem, Vault, etc)

Reply:
this is how i modeled in the POC https://github.com/newrelic/newrelic-auth-rs/pull/20/files#diff-61632ab053201971c7f04dcc71c92c6aaf19c42f0481a3cd377215fd429dcadbR10

Reply:
Ok, let me rework this output then. Are there any other possible outputs? Would `Option<Vec<u8>>` be a possibility here?

Reply:
Using the API you can also update an existing Identity, so you could add a public key to an identity that has a secret, but I don't think we need to support this use case. I like the idea of @gsanchezgavier with the enum.